### PR TITLE
fix: helm charts have the potential to clobber each other and don't match repo names

### LINKED ISSUE #1765: Helm chart clobbering within components
### Steps to reproduce

Given the following `zarf.yaml` in a directory side by side w/ the Zarf repo:

```yaml
kind: ZarfPackageConfig
metadata:
  name: scratch
  version: 0.0.1

components:
  - name: helm-charts-clobbering
    required: false
    charts:
      - name: podinfo
        localPath: ../zarf/examples/helm-charts/chart
        namespace: chart-1
        version: 6.3.5
      - name: oci-demo
        version: 6.3.5
        namespace: chart-2
        url: oci://ghcr.io/stefanprodan/charts/podinfo
    images:
      - ghcr.io/stefanprodan/podinfo:6.3.5
```

1. `zarf package create . --confirm`
2. `zarf tools archiver decompress zarf-package-scratch-amd64-0.0.1.tar.zst unpacked --unarchive-all`
3. `tree unpacked/components`

### Expected result

2 charts shown

### Actual Result

```
unpacked/components
└── helm-charts-clobbering
    └── charts
        └── oci-demo-6.3.5.tgz
```

Since the 2nd Helm chart is also `podinfo`, it clobbers the first one and destroys the expected `podinfo-6.3.5.tgz`. This results in a broken package as only the final `podinfo` chart will be saved.

### Severity/Priority

IDK, you _really_ shouldn't be using multiple Helm charts with the same _exact_ name and version within a single component (kinda an antipattern tbh), so not really sure where this falls. Opened the issue for tracking.

### Additional Context
Add any other context or screenshots about the technical debt here.

### LINKED ISSUE #2062: zarf prepare find-images is not consistent between OCI and Helm chart registries
### Environment
Device and OS:
- NAME="Ubuntu"
- VERSION="20.04.5 LTS (Focal Fossa)"
App version: v0.30.0
Kubernetes distro being used: None
Other: CLI usage

### Steps to reproduce
1. Execute 

```sh
zarf prepare find-images -l debug --kube-version=1.23.0-0
```

```yaml
kind: ZarfPackageConfig
metadata:
  name: helm-charts
  description: Example showcasing mutiple ways to deploy helm charts
  version: 0.0.1
components:
  - name: demo-helm-oci-chart
    charts:
      - name: asdasd
        version: 6.4.0
        namespace: podinfo-from-oci
        url: oci://ghcr.io/stefanprodan/charts/podinfo
  - name: demo-helm-alt-release-name
    charts:
      - name: podinfo
        version: 6.4.0
        namespace: helm-alt-release-name
        url: https://stefanprodan.github.io/podinfo/podinfo
        releaseName: cool-name
```

The list of images is gathered correctly, even though the name from the demo-helm-oci-chart component is being wrong.

Later Edit:

If I change the component  demo-helm-oci-chart name to asdasd/podinfo then the command fail and the name is taken into consideration:

```yaml
kind: ZarfPackageConfig
metadata:
  name: helm-charts
  description: Example showcasing mutiple ways to deploy helm charts
  version: 0.0.1
components:
  - name: demo-helm-oci-chart
    charts:
      - name: asdasd/podinfo
        version: 6.4.0
        namespace: podinfo-from-oci
        url: oci://ghcr.io/stefanprodan/charts/podinfo
  - name: demo-helm-alt-release-name
    charts:
      - name: podinfo
        version: 6.4.0
        namespace: helm-alt-release-name
        url: https://stefanprodan.github.io/podinfo/podinfo
        releaseName: cool-name
```
Logs:

```text
  DEBUG   2023-10-10T12:39:42Z  -  Reading YAML at zarf.yaml
  DEBUG   2023-10-10T12:39:43Z  -  rename /tmp/zarf-1580237310/components/demo-helm-oci-chart/charts/podinfo-6.4.0.tgz /tmp/zarf-1580237310/components/demo-helm-oci-chart/charts/asdasd/podinfo-6.4.0.tgz: no such file or directory
     ERROR:  Unable to save the chart tarball
```

### Expected result

Both components should use the name key or have consistency between their configs.

### Actual Result

On the demo-helm-oci-chart component the name is taken into consideration but on the demo-helm-alt-release-name component based on OCI registry is not.

### LINKED ISSUE #2063: localPath for charts won't accept tarball
### Environment
Device and OS: M1 Macbook Pro
App version: v0.28.4
Kubernetes distro being used: n/a
Other:

### Steps to reproduce
Just create a zarf package that leverages the charts component and point the localPath attribute at the chart tarball

### Expected result
I would expect the above to work as it does for the Helm CLI

### Actual Result
The zarf package create command fails validation stating that it was unable to package the chart: Chart.yaml file is missing

(I appreciate that zarf is trying to validate my chart, but I can validate my own chart if it means that I can avoid needing to decompress my chart)

### Visual Proof (screenshots, videos, text, etc)
In this component, I have the hack I'm using to get around this commented out, but it would be nice if I didn't need to:

```yaml
components:
  - name: local
    required: false
    charts:
      - name: cert-manager
        namespace: test
        localPath: charts/cert-manager-v1.12.2.tgz
        # localPath: charts/cert-manager
        version: v1.12.2
        valuesFiles:
          - bundles/values.yaml
    # actions:
    #   onCreate:
    #     before:
    #       - dir: charts
    #         cmd: tar xf cert-manager-v1.12.2.tgz
    #     after:
    #       - dir: charts
    #         cmd: rm -r cert-manager ; exit 0
```

### Additional Context
If I have overlooked a way to get this to work, my apologies, but I searched around and nothing jumped out right away.

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This adds support for local charts from tarballs instead of only directories.

## Related Issue

Fixes #2062 
Fixes #1765

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/internal/packager/helm/repo.go
Comment (Noxsios) [inline src/internal/packager/helm/repo.go:1]: please update `src/types/component.go`'s `ZarfChart.LocalPath`'s jsonschema description

**Review by Noxsios — CHANGES_REQUESTED** [review]

File: src/internal/packager/helm/repo.go
Comment (Racer159) [reply src/internal/packager/helm/repo.go:1]: done

**Review by Noxsios — APPROVED** [review]
lgtm

**Comment by Racer159** [comment]
Moving back to a draft to work more on addressing #1765 - this would require indexing charts like files and data injections.

**Comment by Racer159** [comment]
Breaking changes:

1. Chart names must match between `chart.name` and the name in the `Chart.yaml`
2. Charts are now held inside of an indexed folder instead of the root of the component

**Pull request closed without merging**